### PR TITLE
refactor anchor generator

### PR DESCRIPTION
**Pull request by zhangxiaoli73**
## What changes were proposed in this pull request?

Here, we generate anchors without change area. Original one will change area and has been deperated in new pytorch code.

## How was this patch tested?

Unit tests

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Anchor.scala [hunk 1]
```diff
@@ -45,6 +46,36 @@ class Anchor(ratios: Array[Float], scales: Array[Float]) extends Serializable {
     getAllAnchors(shiftX, shiftY, basicAnchors)
   }
 
+  /**
+   * Here, we generate anchors without change area.
+   * Original one will change area and has been deperated in new pytorch code.
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Anchor.scala]: No need this comment

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Anchor.scala [hunk 2]
```diff
@@ -123,27 +154,27 @@ class Anchor(ratios: Array[Float], scales: Array[Float]) extends Serializable {
...
-      while (j <= scaleAnchors.size(1)) {
-        anchors.update(idx, scaleAnchors(j))
-        idx = idx + 1
-        j += 1
-      }
-      i += 1
-    }
-    anchors
-  }
+//  private[nn] def generateBasicAnchors(_ratios: Array[Float], _scales: Array[Float],
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Anchor.scala]: Remove commented code ?

Comment (zhangxiaoli73) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Anchor.scala]: ok,changed

**Comment by zhangxiaoli73** [comment]
jenkins pass: http://mlt-bddl01.sh.intel.com:8080/view/PR/job/BigDL-PRVN-lite/1035/

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/nn/RegionProposalSpec.scala [hunk 3]
```diff
@@ -265,10 +265,11 @@ class RegionProposalSpec extends FlatSpec with Matchers {
     layer.evaluate()
     val output = layer.forward(T(T(features), images)).toTable
     val outputExpected = Tensor[Float](
-      T(T(0.0f, 0.0f, 20.999596f, 19.0f),
-      T(0.0f, 0.0f, 12.995603f, 19.0f),
-      T(0.0f, 0.0f, 37.0f, 19.0f),
-      T(0.0f, 0.0f, 29.011127f, 13.003019f)
+      T(T(0.0f, 0.0f, 19.313313f, 19.0f),
```
Comment (i8run) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/nn/RegionProposalSpec.scala]: Why you changed the expected output ? Does it means, there's a bug of previous implementation before this commit?

Comment (zhangxiaoli73) [reply spark/dl/src/test/scala/com/intel/analytics/bigdl/nn/RegionProposalSpec.scala]: yes, original nms implements have been changed in lastest pytorch as it is not exactly same with its meaning. So I also changed nms in bigdl and it's expectation.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Anchor.scala [hunk 4]
```diff
@@ -45,6 +46,35 @@ class Anchor(ratios: Array[Float], scales: Array[Float]) extends Serializable {
     getAllAnchors(shiftX, shiftY, basicAnchors)
   }
 
+  /**
+   * Here, we generate anchors without change area.
+   * @param ratios
+   * @param scales
+   * @param baseSize
+   * @return
```
[the remark this answers is not in the document]
Comment (zhangxiaoli73) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Anchor.scala]: ok, changed

**Comment by zhangxiaoli73** [comment]
jenkins pass：
http://mlt-bddl01.sh.intel.com:8080/job/BigDL-PRVN-lite/1067/

**Pull request merged**